### PR TITLE
Plugin Management: add a delay for some plugin actions

### PR DESCRIPTION
**Pull request by yashwin**
#### Proposed Changes

Plugins are filtered based on the `status` key, which made it disappear from the current tab and confused the user about what exactly happened. To show a clear status of the action like `Activated`, `Deactivated` etc, before the plugin is moved to a different tab, we are delaying the process of updating the `status` by 3 seconds. Also, to show the current status, we have added a new key called `displayStatus` to show the actual status on the UI.

- Add some delay(3000ms) before the element disappears from the list when performing different plugin actions.
- Use `displayStatus` if available to show the status in the UI.

#### Testing Instructions

> **_NOTE:_** You can test these changes in Calypso Blue or Green.

**Prerequisites**

Since this change is made specifically for agencies, you must set yourself(partner) as an agency if you want to test in Calypso Green - 2c49b-pb. Make sure to switch it back to the previous type. 

**Instructions**

1. Run `git checkout update/add-delay-to-update-plugin-action-status` and `yarn start-jetpack-cloud` or `yarn start` if you want to test these changes in Calypso Blue.
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard -> Click on **Plugins** in the sidebar or visit http://calypso.localhost:3000/plugins/manage if you want to test these changes in Calypso Blue.
3. Click on the `Active` tab -> Click on the toggle for `Active` on any site. -> Verify that the `Deactivated` status shows below the plugin, and the plugin disappears after 3 seconds.

**Recording**

https://user-images.githubusercontent.com/10586875/194531006-6d498f75-a331-4648-9752-db666eaeefec.mov

4. Test the same for actions like activating a plugin, deactivating a plugin & updating a plugin, and verify it works as shown in the recording.
5. Test all the above actions with bulk actions and verify it works as shown in the recording.

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into the trunk. Inapplicable items can be left unchecked.

The PR author and reviewer are responsible for completing the checklist.

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] ~~Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?~~
- [x] Have you checked for TypeScript, React, or other console errors?
- [x] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [x] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to 1202619025189113-as-1202971374349855

### CODE REVIEW

File: client/my-sites/plugins/plugin-management-v2/plugin-action-status/index.tsx [hunk 1]
```diff
@@ -24,7 +24,7 @@ export default function PluginActionStatus( {
 }: Props ) {
 	// Group statuses by status type(completed, error, inProgress)
 	const groupedStatues = currentSiteStatuses.reduce( ( group, plugin ) => {
-		const { status } = plugin;
+		const status = plugin.actualStatus || plugin.status;
```
Comment (rcanepa) [inline client/my-sites/plugins/plugin-management-v2/plugin-action-status/index.tsx]: What's the difference between these two fields?

Comment (yashwin) [reply client/my-sites/plugins/plugin-management-v2/plugin-action-status/index.tsx]: I have renamed it to `displayStatus`. Basically this is used only to show the status on the UI. Plugins are filtered based on the `status`.

File: client/state/plugins/installed/test/actions.js [hunk 2]
```diff
@@ -258,13 +258,7 @@ describe( 'actions', () => {
...
 		test( 'should dispatch plugin activate request success action when request completes', async () => {
 			await activatePlugin( 2916284, { slug: 'akismet', id: 'akismet/akismet' } )( spy, getState );
-			expect( spy ).toHaveBeenCalledWith( {
-				type: PLUGIN_ACTIVATE_REQUEST_SUCCESS,
-				action: ACTIVATE_PLUGIN,
-				siteId: 2916284,
-				pluginId: 'akismet/akismet',
-				data: { ...akismet, active: true, log: [ 'Plugin activated.' ] },
-			} );
+			expect( spy ).toHaveBeenCalledWith( expect.any( Function ) );
```
Comment (yashwin) [inline client/state/plugins/installed/test/actions.js]: ~~These changes are due to a function being called on success.~~

**Review by robertf4 — COMMENTED** [review]
Looks really good overall, and the UX is tangibly better than it was.

One thing I noticed is that if I have an active plugin, toggle it to inactive, and quickly switch to the inactive tab then the inactive plugin doesn't show there until the timer ticks and the dispatch fires. At this point the inactive plugin shows as expected.

There is a slight twist on this solution that I think would give the best of both worlds:
1. Instead of a `displayStatus` property on the plugin, add a property called something like `statusRecentlyChanged` which starts as null, `true`, or `false`.
2. Add actions that can be dispatched to change the state of `statusRecentlyChanged` to false/true.
3. Remove the `delayDispatch` property which is being used to toggle this behavior.
4. In `handleDispatchSuccessCallback` immediately dispatch the regular action, and also immediately dispatch an action which sets `statusRecentlyChanged` to `true`. Also call a `setTimeout` which will set `statusRecentlyChanged` to `false` in 3 seconds.
5. In the `_filters` in `client/state/plugins/installed/selectors.js`, return any plugin which matches the current filter as well as any plugin with `statusRecentlyChanged` set to `true`.

I haven't tried this to ensure all the details work out, but I think something like this will work. Basically the goal is not to delay changing the plugin status, but instead make it show up in the prior status for just a little while. Thus it will display on the tab of its current status as well as the tab it was recently on. It also might help for future functionality which could on the status to always be up-to-date.

What do you think?

**Comment by yashwin** [comment]
@robertf4 Thanks for the review and the approach you suggested. I like the approach, and I'll see if this will work.

**Comment by yashwin** [comment]
@robertf4 -  I have made the changes as you suggested.

**Review by robertf4 — APPROVED** [review]
Looks good and works great! 🚢

**Pull request merged**